config/config.go

```diff
@@ -51,12 +51,26 @@ type Adapter struct {
 }
 
 type Metrics struct {
+	GoMetrics  GoMetrics         `mapstructure:"gometrics"`
+	Influxdb   InfluxMetrics     `mapstructure:"influxdb"`
+	Prometheus PrometheusMetrics `mapstructure:"prometheus"`
+}
+
+type InfluxMetrics struct {
 	Host     string `mapstructure:"host"`
 	Database string `mapstructure:"database"`
 	Username string `mapstructure:"username"`
 	Password string `mapstructure:"password"`
 }
 
+type PrometheusMetrics struct {
+	Enabled string `mapstructure:"enabled"`
+}
+
+type GoMetrics struct {
+	Enabled string `mapstructure:"enabled"`
```

Comment:
I'm very confused by this... it seems like `go-metrics` should be an implementation detail. It helps for _some_ backends (like `influx`), but hurts for others (like `prometheus`). Why has this bubbled all the way out to the `config`?

The only use I see of this variable is in `NewMetricsEngine`... and it looks very suspicious:

> // Seperate check in case we find another metrics library we want to hook to InfluxDB
> // If we want to support tagging, then we will need something other than the legacy goMetrics

Seems much simpler, more flexible, and less error-prone if the `config` just offers backends, and the code gets to decide when `go-metrics` is appropriate.